shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rule/EncryptTable.java

```diff
@@ -48,7 +48,7 @@ public EncryptTable(final EncryptTableRuleConfiguration config, final Map<String
             columns.put(each.getLogicColumn(), new EncryptColumn(getEncryptColumnDataType(each.getLogicDataType(), dataTypes), each.getCipherColumn(),
                     getEncryptColumnDataType(each.getCipherDataType(), dataTypes), each.getAssistedQueryColumn(), getEncryptColumnDataType(each.getAssistedQueryDataType(),
                             dataTypes),
-                    each.getPlainColumn(), getEncryptColumnDataType(each.getPlainDataType(), dataTypes), each.getEncryptorName()));
+                    each.getPlainColumn(), getEncryptColumnDataType(each.getPlainDataType(), dataTypes), each.getEncryptorName(), each.getQueryWithCipherColumn()));
```

Comment:
How about inherit the `isQueryWithCipherColumn` on table with column here?

Reply:
I think that's ok，  because I have modified  `org.apache.shardingsphere.encrypt.rule.EncryptTable#getQueryWithCipherColumn(java.lang.String)`

Reply:
OK, can you add some UTs to ensure that the inheritance order on `isQueryWithCipherColumn` is global->table->column, and the `isQueryWithCipherColumn` priority is column > table > global.

Reply:
In addition, could you add some test to ensure  that the `isQueryWithCipherColumn ` on column compare to the `isQueryWithCipherColumn ` on table, the performance was not reduced.

Reply:
I added some test cases for `isQueryWithCipherColumn`.
What do you recommend for performance testing?

Reply:
Can you provide your wechat id, I will add you into our kernel group.

Reply:
guhong06